src/Raven.Client/Json/Converters/BlittableJsonReaderArrayConverter.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+using Sparrow.Json;
+
+namespace Raven.Client.Json.Converters
+{
+    internal sealed class BlittableJsonReaderArrayConverter : RavenTypeJsonConverter<BlittableJsonReaderArray>
+    {
+        public static readonly BlittableJsonReaderArrayConverter Instance = new BlittableJsonReaderArrayConverter();
+
+        private BlittableJsonReaderArrayConverter() {}
+
+        protected override void WriteJson(BlittableJsonWriter writer, BlittableJsonReaderArray value, JsonSerializer serializer)
+        {
+            writer.WriteStartArray();
+            foreach (var item in value)
+            {
+                serializer.Serialize(writer, item);
+            }
+            writer.WriteEndArray();
+        }
+
+        internal override BlittableJsonReaderArray ReadJson(BlittableJsonReader blittableReader)
+        {
+            if (!(blittableReader.Value is BlittableJsonReaderArray blittableArrayValue))
+            {
+                throw new SerializationException(
+                    $"Can't convert {blittableReader.Value.GetType()} type to {nameof(BlittableJsonReaderArray)}. The value must to be an array");
+            }
+            
+            //Skip in order to prevent unnecessary movement inside the blittable array
+            blittableReader.SkipBlittableArrayInside();
```

Comment:
I don't understand this comment

Reply:
https://github.com/ravendb/ravendb/pull/7495#discussion_r220083567

Reply:
done